src/Resources/skeleton/authenticator/LoginFormEntityAuthenticator.tpl.php

```diff
@@ -0,0 +1,86 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace ?>;
+
+use <?= $user_fully_qualified_class_name ?>;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\Exception\InvalidCsrfTokenException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class <?= $class_name; ?> extends AbstractFormLoginAuthenticator
+{
+    use TargetPathTrait;
+    private $entityManager;
+    private $router;
+    private $csrfTokenManager;
+    private $passwordEncoder;
+
+    public function __construct(EntityManagerInterface $entityManager, RouterInterface $router, CsrfTokenManagerInterface $csrfTokenManager, UserPasswordEncoderInterface $passwordEncoder)
+    {
+        $this->router = $router;
+        $this->csrfTokenManager = $csrfTokenManager;
+        $this->passwordEncoder = $passwordEncoder;
+        $this->entityManager = $entityManager;
+    }
+
+    public function supports(Request $request)
+    {
+        return 'app_login' === $request->attributes->get('_route')
+            && $request->isMethod('POST');
+    }
+
+    public function getCredentials(Request $request)
+    {
+        $credentials = [
+            'email' => $request->request->get('email'),
```

Comment:
Hmm. Another tricky spot. What if the user is actually using a `username` property? Dang these little details! In `MakeUser`, this is something we ask: what is your "identifier" property - e.g. `email` or `user`. When this command is ran, I don't think we know/can guess that info. I think we will need to ask them another question. I'm not totally sure how to phrase it:

> When field on your `User` class will people enter when logging in?

I guess if their `User` class has an `email` or `username` property (but not both), it's safe to guess. But, it *does* need to be configurable I think. Which, affects a few other places.

Who thought generators would be so hard? ;)

Reply:
ok, that was actually a bit "brain-fuck" ;)
i addition with the previous comment, i had too much authenticator templates, so i've merged all them into one dynamic template.
I'm also automatically guessing username field when the provider is an "entity" type, then we got the field